docs/framework/guides/tutorials/data-from-external-source.md

```diff
@@ -0,0 +1,877 @@
+---
+category: framework-tutorials
+order: 10
+---
+
+# Data from external source
+
+In this tutorial, you will learn how to implement a widget that fetch data from an external source and updates all own instances in set interval of time.
+
+You will build an "external data fetch" feature that allows the users to insert predefined widget that will show current Bitcoin rate and it will be updated with the set time interval fetching data from predefined external source. You will use widget utilities and conversion in order to define the behavior of this feature. Later on, you will use button utilities to create a button that will allow for inserting new instances of external source widgets. You will also learn how to update the widget data based on the editor API.
```

Comment:
```suggestion
You will build an "external data fetch" feature that allows users to insert a predefined widget that will show the current Bitcoin rate and it will be updated with the set time interval fetching data from a predefined external source. You will use widget utilities and conversion in order to define the behavior of this feature. Later on, you will use button utilities to create a button that will allow for inserting new instances of external source widgets. You will also learn how to update the widget data based on the editor API.
```

Reply:
My gut tells me this paragraphs repeats the first paragraph of the guide. Maybe we can get rid of the first paragraph then?

Reply:
> use button utilities

I'd call it a UI library. And link to the guide [https://ckeditor5.github.io/docs/nightly/ckeditor5/latest/framework/guides/architecture/ui-library.html.](https://ckeditor5.github.io/docs/nightly/ckeditor5/latest/framework/guides/architecture/ui-library.html.)

I couldn't find a reference to the `ButtonView` in the API in the whole guide either.

Reply:
> I couldn't find a reference to the `ButtonView` in the API in the whole guide either.

Correct, now i've update it.